case_test.go

```diff
@@ -0,0 +1,110 @@
+package squirrel
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCaseWithVal(t *testing.T) {
+	caseStmt := Case().
+		What("number").
```

Comment:
If you think so :) I was considering this approach, but decided that for developer it's unclear that `Case(args ...interface{})` expects exactly 0 or 1 argument. 
Also, the question is what to do when it receives more than 1 arg? Ideally, it should return an error, but it breaks interface.